core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -353,6 +358,7 @@ public boolean joiningCluster() {
     private void innerJoinCluster() {
         DiscoveryNode masterNode = null;
         final Thread currentThread = Thread.currentThread();
+        nodeJoinController.startAccumulatingJoins();
```

Comment:
since the election process is async, it may be that other nodes finish it before we do. It would be a shame to loose their joins. This can't really hurt because 99% of the times, an incoming join will mean the local node will become a master soon and will process it correctly.